### PR TITLE
[BACKLOG-39572][UPDATE][Java17Compatibility]

### PR DESCRIPTION
**Pull request by mmujahiduddin**
BACKLOG-39572 UPDATE Java17Compatibility

### CODE REVIEW

File: editor/pom.xml [hunk 1]
```diff
@@ -67,5 +67,29 @@
       <classifier>tests</classifier>
       <scope>test</scope>
     </dependency>
+    <dependency>
```
Comment (peterrinehart) [inline editor/pom.xml:70]: Is this dependency needed?  If we weren't using it before, we don't seem to be now, either.

File: editor/pom.xml [hunk 2]
```diff
@@ -67,5 +67,29 @@
...
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>net.bytebuddy</groupId>
+      <artifactId>byte-buddy-agent</artifactId>
+      <version>1.12.16</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>net.bytebuddy</groupId>
```
Comment (peterrinehart) [inline editor/pom.xml:83]: If this built and ran fine without the byte-buddy and objenesis dependencies, let's leave them out.  Maven should get them on its own, but sometimes doesn't.

**Review by peterrinehart — CHANGES_REQUESTED** [review]

File: editor/pom.xml (hunk 1 above)
Comment (mmujahiduddin) [reply editor/pom.xml:70]: yes without this it gives initializationError, java.lang.NoClassDefFoundError: org/hamcrest/SelfDescribing
Caused by: java.lang.ClassNotFoundException: org.hamcrest.SelfDescribing

File: editor/pom.xml (hunk 2 above)
Comment (mmujahiduddin) [reply editor/pom.xml:83]: yeah byte-buddy is required otherwise we get java.lang.IllegalStateException and removal of objenesis passes

**Pull request merged**